src/functions/docs-feedback.js

```diff
@@ -0,0 +1,356 @@
+  const {
+    updateDiscussion: {
+      discussion: { body: updatedBody },
+    },
+  } = await queryGitHub(
+    dedent(`
+      mutation UpdateDiscussion($discussionId: ID!, $body: String!) { 
+        updateDiscussion(input: {
+          discussionId: $discussionId,
+          body: $body,
+        }) {
+          discussion {
+            body
+          }
+        }
+      }
+    `),
+    {
+      variables: {
+        discussionId: id,
+        body: updateRating(currentBody, rating),
+      },
+    }
+  );
+  console.info('... done!', 'Updated body:\n', updatedBody);
+}
+
+async function reOpenDiscussion({ id }) {
+  console.info('Re-opening discussion...');
+  const {
+    reopenDiscussion: {
+      discussion: { closed },
+    },
+  } = await queryGitHub(
+    dedent(`
+      mutation ReopenDiscussion($discussionId: ID!) {
+        reopenDiscussion(input: {
+          discussionId: $discussionId
+        }) {
+          discussion {
+            closed
+          }
+        }
+      }
+    `),
+    {
+      variables: {
+        discussionId: id,
+        closed: false,
+      },
+    }
+  );
+  console.info('... done!');
+}
+
+async function addDiscussionComment({ id, received, rating, comment }) {
+  console.info('Adding comment to discussion...');
+  const {
+    addDiscussionComment: {
+      comment: { body: addedComment, url },
+    },
+  } = await queryGitHub(
+    dedent(`
+      mutation AddDiscussionComment($discussionId: ID!, $body: String!) {
+        addDiscussionComment(input: {
+          discussionId: $discussionId,
+          body: $body,
+        }) {
+          comment {
+            body
+            url
+          }
+        }
+      }
+    `),
+    {
+      variables: {
+        discussionId: id,
+        body: createBody({ ...received, rating, comment }),
+      },
+    }
+  );
+  console.info('... done!, Added comment:', '\n', url, '\n', addedComment);
+  return url;
+}
+
+async function createDiscussion({ received, path, rating, comment, title }) {
+  console.info(`Creating new discussion for ${path}...`);
+  const {
+    createDiscussion: {
+      discussion: { title: addedTitle, body: addedBody, url },
+    },
+  } = await queryGitHub(
+    dedent(`
+      mutation CreateDiscussion($repositoryId: ID!, $categoryId: ID!, $title: String!, $body: String!) { 
+        createDiscussion(input: {
+          repositoryId: $repositoryId,
+          categoryId: $categoryId,
+          title: $title,
+          body: $body
+        }) {
+          discussion {
+            title
+            body
+            url
+          }
+        }
+      }
+    `),
+    {
+      variables: {
+        repositoryId,
+        categoryId,
+        title,
+        body: createBody({ isFirst: true, ...received, rating, comment }),
+      },
+    }
+  );
+  console.info('... done!, Added discussion:', '\n', url, '\n', addedTitle, '\n', addedBody);
+
+  return url;
+}
+
+const requestsCache = {};
```

Comment:
How does this work? The function gets destroyed after handling the request so, this object will always be empty. No?

Reply:
Okay so it looks like stuff outside the handler does persist (for an instance) But you could also have multiple instances. And the cache will reset every time the functions is respawned. I guess that's probably fine for now.

https://lihbr.com/posts/rate-limiting-without-overhead-netlify-or-vercel-functions

Reply:
Definitely a TIL moment

Reply:
Thanks for that link! It verified my approach and taught me that I can get the ip from the function event's headers instead of doing so client-side, which is _much_ simpler.